OpenSim/Simulation/StatesTrajectory.h

```diff
@@ -155,6 +155,16 @@ class Model;
  *               << std::endl;
  * }
  * @endcode
+ *
+ * Here's an example of iterating from a state at `timeA` to the last state
+ * before `timeB`:
+ * @code{.cpp}
+ * for (const auto& state : SimTK::makeIteratorRange(
+ *                              states.findAt(timeA),
+ *                              states.findNearestBefore(timeB)) {
+ *     ...
```

Comment:
This is  very helpful and likely what Tools will adopt to cycle through available states.